src/core_modules/capture-core/utils/reactQueryHelpers/query/useMetadataQuery.js

```diff
@@ -38,3 +50,19 @@ export const useIndexedDBQuery = <TResultData>(
                 throwErrorForIndexedDB(error);
             },
         });
+
+export const useMetadataApiQuery = <TResultData>(
+    queryKey: QueryKey,
+    queryObject: ResourceQuery,
+    queryOptions?: UseQueryOptions<TResultData>,
+): Result<TResultData> => {
+    const dataEngine = useDataEngine();
+    const queryFn: QueryFunction<TResultData> = () => dataEngine.query({ theQuerykey: queryObject })
+        .then(response => response.theQuerykey);
+    return useAsyncMetadata(queryKey, queryFn, {
+        // TODO: What do you think is sensible here?
+        cacheTime: Infinity,
+        staleTime: Infinity,
```

Comment:
When do we want to refetch metadata from the API?

Reply:
I think `Infinity` is fine.